hack/check-conflicts.sh

```diff
@@ -0,0 +1,41 @@
+#!/bin/bash -e
+
+echo -e "\nChecking for presence of conflict notes in source files..."
+
+overall_failed=0
+
+echo -ne "\n\tChecking Staged files... "
+if [ $(git diff --cached --name-only | wc -l) != "0" ]; then
+    result=$(git diff --cached --name-only | grep -v "vendor/" | xargs grep -PiIrnH '<<<<''<<<|>>>''>>>>|===''====' || true)
+    if [ -z "$result" ]; then
+        echo PASS
+    else
+        echo FAIL
+        echo ""
+        echo "$result"
+        echo ""
+        overall_failed=1
+    fi
+else
+    echo PASS
+fi
```

Comment:
why checking files on stagging, when is this useful? We do not care if there are conflicts until they are checked into the repo, rigth?

Reply:
It is useful for checking the new files you are about to commit (that are not tracked yet) to avoid accidental commits.

Reply:
ok, make sense - would you mind to wrap then these check in a function? you perform the identical check two times, once on staged files, another time on files in repo?

Reply:
@pedjak Fixed that, thx